pandas/tests/series/methods/test_append.py

```diff
@@ -61,6 +61,20 @@ def test_append_tuples(self):
 
         tm.assert_series_equal(expected, result)
 
+    def test_append_dataframe(self):
+        # GH 30975
+        df = pd.DataFrame({"A": [1, 2], "B": [3, 4]})
+        df2 = pd.DataFrame({"C": [5, 6], "D": [7, 8]})
+
+        expected = pd.Series(pd.concat([df.A, df2.D]))
+        result = df.A.append(df2.D)
+
+        tm.assert_series_equal(expected, result)
+
+        msg = "to_append should be a Series or list/tuple of Series, got " \
+              "<class 'pandas.core.frame.DataFrame'>"
+        with pytest.raises(TypeError, match=msg):
+            df.A.append(df)
```

Comment:
also need to add test to check items in a list/tuple

Reply:
Is this an acceptable change:

```suggestion
df = pd.DataFrame({"A": [1, 2], "B": [3, 4]})
li = [df.B, df]
msg = "to_append should be a Series or list/tuple of Series, got list of DataFrame"
with pytest.raises(TypeError, match=msg):
    df.A.append(li)
```